docs/devguide/quality_of_service.rst

```diff
@@ -120,1047 +103,881 @@ The following examples illustrate how to obtain the default policies for publish
 
-       ``service_cleanup_delay.nanosec``
+       - ``DURATION_INFINITE_SEC``
 
-     - ``VOLATILE_DURABILITY_QOS``
+         ``DURATION_INFINITE_NSEC``
 
-       ``DURATION_ZERO_SEC``
+  Specification: :omgspec:`dds:2.2.3.16 LIFESPAN`
 
-       ``DURATION_ZERO_NSEC``
+The default value of the ``duration`` member is infinite, which means samples never expire.
 
-   * - ``DEADLINE``
+.. note::
 
-     - ``period.sec``
+  OpenDDS currently supports expired sample detection on the publisher side when using a :ref:`qos-durability` ``kind`` other than :ref:`VOLATILE_DURABILITY_QOS <VOLATILE_DURABILITY_QOS>`.
+  The current OpenDDS implementation may not remove samples from the data writer and data reader caches when they expire after being placed in the cache.
 
-       ``period.nanosec``
+The value of this policy may be changed at any time, but only affects data written after the change.
 
-     - ``DURATION_INFINITE_SEC``
+.. _qos-user-data:
+.. _quality_of_service--user-data:
 
-       ``DURATION_INFINITE_NSEC``
+User Data QoS
+=============
 
-   * - ``LATENCY_BUDGET``
+..
+    Sect<3.2.11>
 
-     - ``duration.sec``
+The user data QoS policy can be used to attach arbitrary information to the created :term:`entity`.
+This policy applies to the domain participant, data reader, and data writer entities via the ``user_data`` member of their respective QoS structures.
 
-       ``duration.nanosec``
+.. important::
 
-     - ``DURATION_ZERO_SEC``
+  This policy is :ref:`mutable <qos-changing>` and does not affect association.
 
-       ``DURATION_ZERO_NSEC``
+  IDL:
 
-   * - ``LIVELINESS``
+  .. code-block:: omg-idl
 
-     - ``kind``
+    struct UserDataQosPolicy {
+      sequence<octet> value;
+    };
 
-       ``lease_duration.sec``
+  .. list-table::
+     :header-rows: 1
 
-       ``lease_duration.nanosec``
+     * - :term:`Applicable entities <Entity>`
 
-     - ``AUTOMATIC_LIVELINESS_QOS``
+       - Members
 
-       ``DURATION_INFINITE_SEC``
+       - :ref:`Default values <qos-defaults>`
 
-       ``DURATION_INFINITE_NSEC``
+     * - :term:`DomainParticipant`, :term:`DataWriter`, :term:`DataReader`
 
-   * - ``RELIABILITY``
+       - ``value``
 
-     - ``kind``
+       - (empty sequence)
 
-       ``max_blocking_time.sec``
+  Specification: :omgspec:`dds:2.2.3.1 USER_DATA`
 
-       ``max_blocking_time.nanosec``
+The value of the policy is available in respective :ref:`built-in topic data <bit>`.
+The value's use is defined by the application.
+For example, the application could attach security credentials via the policy that can be used by the remote application to authenticate the source. [#user_data_ex]_
 
-     - ``BEST_EFFORT_RELIABILITY_QOS``
+.. warning::
 
-       ``DURATION_INFINITE_SEC``
+  When using :ref:`dds_security`, the user data of a participant can be leaked in unsecured discovery messages.
+  Enabling ``[rtps_discovery]SecureParticipantUserData`` will only send and provide the real user data when it can be securely sent.
```

Comment:
This looks strange in the rendered document.  The `[rtps_discovery]SecureParticipantUserData` should look like it appears in the ini format, or just link to the other section.

Reply:
This is what what the rendered config domain links look like. [Here is the example RST](https://github.com/OpenDDS/OpenDDS/blob/e4e9498857e3a025a8665cfc0f933e4147633a17/docs/internal/docs.rst?plain=1#L565) for the config domain. Right now the links looks the same as the input expect for sections discriminators appear like ``[transport] default_address (rtps_udp)``.

Reply:
Maybe this is something we can address in a follow-on PR?